CMakePresets.json

```diff
@@ -0,0 +1,37 @@
+{
+  "version": 1,
+  "configurePresets": [
+    {
+      "name": "base",
+      "hidden": true,
+      "generator": "Ninja",
```

Comment:
Do we want to require `Ninja` as dependency? Often needs a separate module to be loaded and is not available on some systems.

Reply:
If we use presets version `< 3`, then we need to specify a generator. But even if we don't need to specify it, I would still prefer to set `Ninja`, since it seems to handle dependencies better than make, and it's not as verbose. It's always possible to overwrite the settings from the preset, by using the normal CLI arguments.